addons/l10n_ke_edi_tremol/wizard/account_move_send.py

```diff
@@ -0,0 +1,34 @@
+from odoo import _, models, fields, api
+
+
+class AccountMoveSend(models.Model):
+    _inherit = 'account.move.send'
+
+    l10n_ke_edi_warning_message = fields.Text(compute='_compute_l10n_ke_edi_warning_message')
+
+    @api.depends('move_ids')
+    def _compute_l10n_ke_edi_warning_message(self):
+        for wizard in self:
+            warning_moves = wizard.move_ids.filtered(lambda m: m.country_code == 'KE' and not m._l10n_ke_check_invoice_ready())
+            if warning_moves:
+                wizard.l10n_ke_edi_warning_message = '\n'.join([
+                    _("The following documents have no details related to the fiscal device, a proforma invoice will be used."),
+                    *(warning_moves.mapped('name'))
+                ])
+            else:
+                wizard.l10n_ke_edi_warning_message = False
+
+    def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data):
+        super()._hook_invoice_document_before_pdf_report_render(invoice, invoice_data)
+        if invoice.country_code == 'KE' and not invoice._l10n_ke_check_invoice_ready():
+            invoice_data['error'] = _(
+                "This document does not have details related to the fiscal device, a proforma invoice will be used."
+            )
+
+    def action_send_and_print(self, from_cron=False, allow_fallback_pdf=False):
+        # Override to allow generating the proforma pdf when the invoice has not been
+        # sent to the fiscal device yet.
+        self.ensure_one()
+        if all(move.country_code == 'KE' for move in self.move_ids):
+            allow_fallback_pdf = True
```

Comment:
I don't like it.
- first, check the country of the company instead of each move separately.
- second, such behavior must be managed by the hooks. You are free to put an error or not and let the pdf be generated or not but I really want to prevent overrides of `action_send_and_print` itself